crates/sui-types/src/committee.rs

```diff
@@ -46,60 +57,100 @@ pub struct Committee {
 
         fp_ensure!(
             // Actual committee size is enforced in sui_system.move.
             // This is just to ensure that choose_multiple_weighted can't fail.
-            voting_rights.len() < u32::MAX.try_into().unwrap(),
+            voting_weights.len() < u32::MAX.try_into().unwrap(),
             SuiError::InvalidCommittee("committee has too many members".into())
         );
 
         fp_ensure!(
-            !voting_rights.is_empty(),
+            !voting_weights.is_empty(),
             SuiError::InvalidCommittee("committee has 0 members".into())
         );
 
         fp_ensure!(
-            voting_rights.iter().any(|(_, s)| *s != 0),
+            voting_weights.iter().any(|(_, s)| *s != 0),
             SuiError::InvalidCommittee(
-                "at least one committee member must have non-zero stake.".into()
+                "at least one committee member must have non-zero voting power.".into()
             )
         );
 
-        voting_rights.sort_by_key(|(a, _)| *a);
-        let total_votes = voting_rights.iter().map(|(_, votes)| *votes).sum();
+        voting_weights.sort_by_key(|(a, _)| *a);
+        
+        let total_votes: VoteUnit = voting_weights.iter().map(|(_, votes)| *votes).sum();
+        fp_ensure!(
+            total_votes == TOTAL_VOTING_POWER,
+            SuiError::InvalidCommittee(
+                format!("total voting power of a committee is {}, must be {}", total_votes, TOTAL_VOTING_POWER)
+            )
+        );
 
-        let (expanded_keys, index_map) = Self::load_inner(&voting_rights);
+        let (expanded_keys, index_map) = Self::load_inner(&voting_weights);
 
         Ok(Committee {
             epoch,
-            voting_rights,
-            total_votes,
+            voting_weights,
             expanded_keys,
             index_map,
             loaded: true,
         })
     }
 
+    /// Normalize the given weights to TOTAL_VOTING_POWER and create the committee
+    pub fn normalize_from_weights(
+        epoch: EpochId,
+        mut voting_weights: BTreeMap<AuthorityName, VoteUnit>,
+    ) -> SuiResult<Self> {
+        fp_ensure!(
+            !voting_weights.is_empty(),
+            SuiError::InvalidCommittee("committee has 0 members".into())
+        );
+        
+        let num_nodes = voting_weights.len();
+        let total_votes: VoteUnit = voting_weights.iter().map(|(_, votes)| *votes).sum();
+        
+        fp_ensure!(
+            total_votes != 0,
+            SuiError::InvalidCommittee(
+                "at least one committee member must have non-zero voting power.".into()
+            )
+        );
+        let normalization_coef = TOTAL_VOTING_POWER as f64 / total_votes as f64;
+        let mut total_sum = 0;
+        for (idx, (_auth, weight)) in voting_weights.iter_mut().enumerate() {
+            if idx < num_nodes - 1 {
+                *weight = (*weight as f64 * normalization_coef).round() as u64; // adjust the weights following the normalization coef
```

Comment:
what if all votes round up? so in practice we lose ~0.5 per node? That will imply the last node might lose significant power, right?